pkg/wasm/convert.go

```diff
@@ -151,7 +151,10 @@ func convert(resource *anypb.Any, cache Cache) (newExtensionConfig *anypb.Any, s
 	}
 	remote := vm.GetCode().GetRemote()
 	httpURI := remote.GetHttpUri()
-	if httpURI == nil {
+	if httpURI == nil || len(httpURI.GetUri()) == 0 {
+		// If the required fields are not set, do not send ECDS push to Envoy,
+		// because Envoy will reject it in the protocol validation.
+		sendNack = true
```

Comment:
So what happens to Envoy? Isn't it permanently un-initialized without ECDS response?

Reply:
Yes. 

Do you think we need to try fetching Wasm somehow in Istio Agent?